readthedocs/telemetry/collectors.py

```diff
@@ -0,0 +1,306 @@
+            Suggests: liblocale-gettext-perl, perl, ecryptfs-utils (>= 67-1)
+            Conffiles:
+            /etc/deluser.conf 773fb95e98a27947de4a95abb3d3f2a2
+            Description: add and remove users and groups
+            This package includes the 'adduser' and 'deluser' commands for creating
+            and removing users.
+            .
+            - 'adduser' creates new users and groups and adds existing users to
+                existing groups;
+            - 'deluser' removes users and groups and removes users from a given
+                group.
+            .
+
+            Package: apt
+            Status: install ok installed
+            Priority: important
+            Section: admin
+            Installed-Size: 4209
+            Architecture: amd64
+            Version: 2.0.6
+            Replaces: apt-transport-https (<< 1.5~alpha4~), apt-utils (<< 1.3~exp2~)
+            Provides: apt-transport-https (= 2.0.6)
+            Depends: adduser, gpgv | gpgv2 | gpgv1, libapt-pkg6.0 (>= 2.0.6)
+
+            Package: base-files
+            Essential: yes
+            Status: install ok installed
+            Priority: required
+            Section: admin
+            Installed-Size: 392
+            Architecture: amd64
+            Multi-Arch: foreign
+            Version: 11ubuntu5.5
+            Replaces: base, dpkg (<= 1.15.0), miscutils
+            Provides: base
+            Depends: libc6 (>= 2.3.4), libcrypt1 (>= 1:4.4.10-10ubuntu3)
+            Pre-Depends: awk
+            Breaks: debian-security-support (<< 2019.04.25), initscripts (<< 2.88dsf-13.3)
+
+        .. note:: Some lines have been removed for brevity.
+        """
+
+        def extract_value(key, line):
+            if line.startswith(key):
+                # pep8 and blank don't agree on having a space before :.
+                return line[len(key) :].strip()  # noqa
+            return ""
+
+        code, stdout, _ = self.run("dpkg", "--status")
+        stdout = stdout.strip()
+        packages = []
+        if code != 0 or not stdout:
+            return packages
+
+        for chunk in stdout.split("\n\n"):
+            lines = chunk.split("\n")
+            package = ""
+            version = ""
+            for line in lines:
+                if not package:
+                    package = extract_value("Package:", line)
+                if not version:
+                    version = extract_value("Version:", line)
+                if package and version:
+                    break
+            # If the version wasn't recovered we
+            # at least got the package name.
+            if package:
+                packages.append(
+                    {
+                        "name": package,
+                        "version": version,
+                    }
+                )
+
+        return packages
+
+    def _get_user_apt_packages(self):
+        return [
+            {"name": package, "version": ""}
```

Comment:
We could get the version with `dpkg --status` for each package, or we could also get it from the list we already have `build["apt"]["all"][user_pacakge]["version"]`